spec.html

```diff
@@ -5403,6 +5403,16 @@ <h1>IsInteger ( _argument_ )</h1>
       </emu-alg>
     </emu-clause>
 
+    <emu-clause id="sec-isnonnegativeinteger" aoid="IsNonNegativeInteger">
+      <h1>IsNonNegativeInteger ( _argument_ )</h1>
+      <p>The abstract operation IsNonNegativeInteger determines if _argument_ is non-negative integer numeric value.</p>
+      <emu-alg>
+        1. If ! IsInteger(_argument_) is *false*, return *false*.
+        1. If _argument_ &ge; 0, return *true*.
+        1. Return *false*.
```

Comment:
I dislike the three cases here being false, true, false. I'd prefer the following:

1. If IsInteger(_argument_) is true and _argument_ &ge; 0, return true.
1. Otherwise, return false.

If you feel strongly about having 3 steps, then I'd prefer false, false, true, i.e. make step 2 check for < 0.

Reply:
TIL, thank you. Fixed in 8552db7.